### ISSUE TITLE
Add endpoint property in WebPushError

### ISSUE THREAD

**Issue by kapoorji** [web-push-libs/web-push#250]
Hi Guys,

thanks for the great library. It would be great if guys can return addition data while throwing error so that it gets handy to catch which subscriber id has failed (unauthorized, not registered.. else)

```
function WebPushError(message, statusCode, headers, body, endpoint) {
  Error.captureStackTrace(this, this.constructor);

  this.name = this.constructor.name;
  this.message = message;
  this.statusCode = statusCode;
  this.headers = headers;
  this.body = body;
  this.endpoint = endpoint;  // added and passing on new object
}
```

**Comment by marco-c** [comment]
Sounds good to me, it definitely doesn't hurt adding it :)

If you want to build a PR, you'd just need to modify https://github.com/web-push-libs/web-push/blob/master/src/web-push-lib.js#L255 and https://github.com/web-push-libs/web-push/blob/master/src/web-push-error.js#L3, and add some tests in https://github.com/web-push-libs/web-push/blob/master/test/testSendNotification.js.

**Comment by marco-c** [comment]
Fixed in #257.
